test/org/javarosa/model/xform/SMSSerializingVisitorTest.java

```diff
@@ -0,0 +1,161 @@
+package org.javarosa.model.xform;
+
+import org.javarosa.core.PathConst;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.transport.payload.ByteArrayPayload;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.javarosa.test.utils.Utils.convertToString;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Performs a comprehensive suite of tests on the results of a form
+ * being serialized to the SMS format.
+ */
+public class SMSSerializingVisitorTest {
+    private FormParseInit formParser;
+    private String SMS;
+    private static final String delimiter = " ";
+    private FormEntryController formEntryController;
+    private FormInstance formInstance;
+
+    @Before
+    public void setup() throws IOException {
+        formParser = new FormParseInit();
+        formParser.setFormToParse(new File(PathConst.getTestResourcePath(), "sms_form.xml").toString());
+        formEntryController = formParser.getFormEntryController();
+        formInstance = formEntryController.getModel().getForm().getInstance();
+
+        SMSSerializingVisitor serializer = new SMSSerializingVisitor();
+
+        ByteArrayPayload payload = (ByteArrayPayload) serializer.createSerializedPayload(formInstance);
+
+        SMS = convertToString(payload.getPayloadStream());
+        SMS = SMS.replace("\\", "").replace("\\\\", "\\");
+    }
+
+    @Test
+    public void SmsNotNull() {
+        assertNotNull(SMS);
+    }
+
+    @Test
+    public void testTagsAreBeingSerialized() {
+        for (String tag : getAnswerTags()) {
+            assertTrue(SMS.contains(tag));
+        }
+    }
+
+    /**
+     * Checks to see if all answers for the selected tags exist. This
+     * targets answers of various data types  and structures such as
+     * groups.
+     */
+    @Test
+    public void testTaggedAnswers() {
+        for (String tag : getAnswerTags()) {
+            assertTrue(taggedAnswerExists(tag, SMS));
+        }
+    }
+
+    /***
+     * Checks to see an answer that doesn't have a tag is present .
+     * Placeholder  is an answer bounded to the <maiden_name> tag
+     */
+    @Test
+    public void testAnswerWithoutATag() {
+        assertFalse(SMS.contains("Placeholder"));
+    }
+
+    /***
+     * Checks to see a tag that doesn't have an answer exists within
+     * the SMS.
+     * the CTY tag is attached to  the country  question.
+     */
+    @Test
+    public void testTagWithNoAnswer() {
+        assertFalse(SMS.contains("CTY"));
+    }
+
+    /**
+     * returns a list of all the tags that are bounded to
```

Comment:
You probably want “bound”, past tense of “bind”. “Bounded” has to do with boundaries.